src/main/java/com/atlassian/jira/cloud/jenkins/logging/PipelineLogger.java

```diff
@@ -16,12 +18,24 @@ public class PipelineLogger {
 
     private static PipelineLogger NOOP_INSTANCE;
 
+    private Boolean debugLogging = Boolean.FALSE;
```

Comment:
Adding a debug mode to the `PipelineLogger` so that `debug()` messages are only logged when this is `true`.